### PR TITLE
Update the CI system

### PR DESCRIPTION
**Pull request by palainp**
Dear @dinosaure,
Thank you for going into qubes-miragevpn :heart: 
This PR is a first try to help with the CI system, based on what is done at qubes-mirage-firewall, it:
- updates opam-repository to a recent state to have the last happy_eyeballs version
- pins two repository that should be removed if PRs can be merged
- changes how the hashsum is compared, and avoid shell shenigans that can be hard to follow, sensible and could fail.
- also update a bit the code, I had some issues.

Unfortunately the hashsum is not the reproducible value as it's based on my local build.
Best.

### CODE REVIEW

**Comment by palainp** [comment]
Oups sorry, I missed the correct branch for this PR, is there a way to target the ci branch now?

File: Dockerfile [hunk 1]
```diff
@@ -27,8 +27,8 @@ RUN opam init --disable-sandboxing -a --bare https://github.com/ocaml/opam-repos
 RUN opam switch create myswitch 4.14.2
 RUN opam exec -- opam install -y mirage opam-monorepo ocaml-solo5
 RUN opam exec -- opam install -y tls
-RUN opam pin https://github.com/palainp/miragevpn.git#update
-RUN opam pin https://github.com/palainp/mirage-qubes.git#update
+RUN opam pin https://github.com/robur-coop/miragevpn.git
+RUN opam pin https://github.com/mirage/mirage-qubes.git
```
Comment (palainp) [inline Dockerfile]: Once mirage-qubes is merged in opam-repository this line should be removed as it will be retrieved with `make depend`.
To me, if miragevpn could also be released in opam-repository, the previous line would also be removed :)

**Comment by dinosaure** [comment]
Ok the error is unrelated to what you did but some of archives are in our git.robur.coop server and we just got few issue on it. So we can not continue this PR until the server is back.

**Comment by palainp** [comment]
> Ok the error is unrelated to what you did but some of archives are in our git.robur.coop server and we just got few issue on it. So we can not continue this PR until the server is back.

Yes, I saw that, take the time you need and be safe :)

File: Dockerfile (hunk 1 above)
Comment (palainp) [reply Dockerfile]: As a side note, to improve reproducibility, I think it's better to add the commit hash if the package is not in opam-repository :)

**Comment by palainp** [comment]
For the time being, I'm keeping the pin on mirage-qubes as it works with the current mirage version (it's pinned as v0.9.X), but it should be removed once mirage-qubes 0.10.0 is merged in opam-repository and a new release for mirage is also integrated.

**Comment by palainp** [comment]
Hmm, the hashsum is not the same and I have a shell syntax error in the action script. I'll fix that quickly, sorry for that :(

**Comment by palainp** [comment]
My local build kept the duniverse folder, I now have the same hashsum as github action :)
